Create read-record-from-the-topic test

Merk1998 opened the pull request:
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Darya Merkureva mdp1998@icloud.com

Merk1998 committed:
Create read-record-from-the-topic test

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Darya Merkureva mdp1998@icloud.com

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/kafka/integration/ReadRecordTest.java:
```diff
@@ -0,0 +1,65 @@
+
+public class ReadRecordTest {
+  private static KafkaNodeContainer kafkaNodeContainer;
+  private MockConsumer<String, String> mockConsumer;
```
please use real consumer instead of the mock for the integration tests

akurilov commented on src/test/java/com/emc/mongoose/storage/driver/kafka/integration/ReadRecordTest.java:
```diff
@@ -0,0 +1,65 @@
+public class ReadRecordTest {
+  private static KafkaNodeContainer kafkaNodeContainer;
+  private MockConsumer<String, String> mockConsumer;
+  private static final String TOPIC_NAME = "test-topic";
```
please use some test local topic name, e.g. "topic" + ReadRecordTest.class.getSimpleName()
different tests may use same kafka instance in future
so we need to avoid the collision

Merk1998 committed:
Changed MockConsumer to KafkaConsumer in Read-Record-From-The-Topic integration test

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Darya Merkureva mdp1998@icloud.com

veronikaKochugova commented on src/test/java/com/emc/mongoose/storage/driver/kafka/integration/ReadRecordTest.java:
```diff
@@ -34,32 +35,44 @@ public static void closeContainer() {
-    beginningOffsets.put(new TopicPartition(TOPIC_NAME, 0), 0L);
-    mockConsumer.updateBeginningOffsets(beginningOffsets);
+    consumer.subscribe(Arrays.asList(TOPIC_NAME));
+    ProducerRecord<String, String> producerRecord =
```
final

veronikaKochugova commented on src/test/java/com/emc/mongoose/storage/driver/kafka/integration/ReadRecordTest.java:
```diff
@@ -34,32 +35,44 @@ public static void closeContainer() {
-        "Record must be " + record.toString(),
-        record,
-        mockConsumer.poll(TIMEOUT).records(topicPartition).get(0));
+    ConsumerRecords<String, String> recordsRead = consumer.poll(TIMEOUT);
```
final

Merk1998 committed:
Change variables recordsRead and producerRecord to final

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Darya Merkureva mdp1998@icloud.com
